noxfile.py

```diff
@@ -46,6 +46,28 @@ def tests(session: nox.Session) -> None:
         *session.posargs,
         env={"COVERAGE_FILE": f".coverage.{session.python}"},
     )
+
+    if session.python == "3.7":
+        # There is nothing more to test with Python 3.7.
+        # Because there is a dependency conflict between
+        # argcomplete and the latest tox (both depend on
+        # a different version of importlib-metadata for Py 3.7)
+        # pip installs tox 3 as the latest one for Py 3.7.
+        # Coverage is < 100 % for Py 3.7 so it's not run by default
+        # unless we test nox with more than one environment.
+        return
+
+    # Test tox_to_nox with older tox
+    session.install("tox<4")
+    session.run(
+        "pytest",
+        "--cov=nox",
+        "--cov-config",
+        "pyproject.toml",
+        "--cov-report=",
+        "tests/test_tox_to_nox.py",
+        env={"COVERAGE_FILE": f".coverage.{session.python}.tox3"},
+    )
```

Comment:
Seems like a good case for `nox.parametrize`? Can parametrize on python 3.7 - 3.11 as well as tox 3 and 4?

Reply:
Tried this myself this morning and I think it's doing what we want, what do you think?:

```python
@nox.session
@nox.parametrize(
    "python, tox",
    [
        (python, tox)
        for python in ("3.7", "3.8", "3.9", "3.10", "3.11")
        for tox in ("3", "4")
        if (python, tox) != ("3.7", "4")
    ],
)
def tests(session: nox.Session, tox: str) -> None:
    """Run test suite with pytest."""
    session.create_tmp()  # Fixes permission errors on Windows
    session.install("-r", "requirements-test.txt")
    session.install(".", "jinja2", f"tox=={tox}")
    session.run(
        "pytest",
        "--cov=nox",
        "--cov-config",
        "pyproject.toml",
        "--cov-report=",
        *session.posargs,
        env={"COVERAGE_FILE": f".coverage.{session.python}"},
    )
    session.notify("cover")
```

Reply:
I had this ready from the previous night and my implementation is little bit different. Take a look at it and let me know which one do you like more.

Reply:
Yeah yours works too 👍🏻 Happy with that